src/Tizen.Security.DevicePolicyManager/Tizen.Security.DevicePolicyManager/DevicePolicyManager.cs

```diff
@@ -0,0 +1,144 @@
+        public DevicePolicyManager()
+        {
+            _handle = Interop.DevicePolicyManager.CreateHandle();
+            DevicePolicyManagerErrorFactory.GetException(ErrorFacts.GetLastResult());
+        }
+
+        /// <summary>
+        /// Method to creates an instance of Device Policy.
+        /// </summary>
+        /// <typeparam name="T">The generic type to create.</typeparam>
+        /// <returns>An instance of policy.</returns>
+        /// <since_tizen> 6 </since_tizen>
+        /// <exception cref="InvalidOperationException">Thrown when failed to create instance of the policy.</exception>
+        public T GetPolicy<T>() where T : class
+        {
+            T policy = Activator.CreateInstance(typeof(T),
+                System.Reflection.BindingFlags.NonPublic | System.Reflection.BindingFlags.Instance,
+                null, new object[] { this }, null) as T;
+
+            if (policy == null)
+            {
+                Log.Error(Globals.LogTag, "Failed to create " + policy.ToString());
+                throw new InvalidOperationException("Failed to create " + policy.ToString());
+            }
+            else
+            {
+                return policy;
+            }
+        }
+
+        internal IntPtr GetHandle()
+        {
+            return _disposed ? IntPtr.Zero : _handle;
+        }
+
+        /// <summary>
+        /// A Destructor of DevicePolicyManager.
+        /// </summary>
+        ~DevicePolicyManager()
+        {
+            this.Dispose(false);
+        }
+
+        /// <summary>
+        /// Releases any unmanaged resources used by this object. Can also dispose any other disposable objects.
+        /// </summary>
+        /// <param name="disposing">If true, disposes any disposable objects. If false, does not dispose disposable objects.</param>
+        protected virtual void Dispose(bool disposing)
+        {
+            if (_disposed)
+            {
+                return;
+            }
+ 
+            if (_handle != IntPtr.Zero)
+            {
+                int ret = Interop.DevicePolicyManager.DestroyHandle(_handle);
+                if (ret != (int)Interop.DevicePolicyManager.DpmError.None)
+                {
+                    Log.Error(Globals.LogTag, "Faild to destroy dpm handle");
+                    throw DevicePolicyManagerErrorFactory.GetException(ret);
+                }
+
+                _handle = IntPtr.Zero;
+            }
+
+            _disposed = true;
+        }
+
+        /// <summary>
+        /// Releases any unmanaged resources used by this object.
+        /// </summary>
+        public void Dispose()
+        {
+            this.Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+    }
+
+    static internal class Globals
+    {
+        internal const string LogTag = "Tizen.Security.DPM";
+    }
+
+    internal static class DevicePolicyManagerErrorFactory
+    {
+        static internal Exception GetException(int error)
```

Comment:
How about this? 
```
static internal void ThrowException(int error) {
    .... 
    switch (errCode)
    {
        case Interop.DevicePolicyManager.DpmError.InvalidParameter:
               throw new ArgumentException("Invalid parameter");
        ....
    }
} 
```

Reply:
I don't prefer it, 
because it always provide a same call stack ends with `DevicePolicyManagerErrorFactory.ThrowException` to app developer when exception was raised